nx_parallel/utils/chunk.py

```diff
@@ -2,65 +2,96 @@
 import os
 import networkx as nx
 
+from nx_parallel.utils.types import GraphIteratorType
+
+
 __all__ = ["chunks", "get_n_jobs", "create_iterables"]
 
 
-def chunks(iterable, n):
-    """Divides an iterable into chunks of size n"""
+def chunks(iterable, n_chunks):
+    """Yield exactly `n_chunks` chunks from `iterable`, balancing the chunk sizes."""
+    iterable = list(iterable)
+    k, m = divmod(len(iterable), n_chunks)
     it = iter(iterable)
-    while True:
-        x = tuple(itertools.islice(it, n))
-        if not x:
-            return
-        yield x
+    for _ in range(n_chunks):
+        chunk_size = k + (1 if m > 0 else 0)
+        m -= 1
+        yield tuple(itertools.islice(it, chunk_size))
 
 
 def get_n_jobs(n_jobs=None):
-    """Returns the positive value of `n_jobs`."""
+    """
+    Return the positive value of `n_jobs`, adjusting for the environment.
+
+    If running under pytest, returns 2 jobs. If using a parallel backend,
+    returns the number of jobs configured for the backend. Otherwise, returns
+    the number of CPUs adjusted for negative `n_jobs` values.
+    """
     if "PYTEST_CURRENT_TEST" in os.environ:
         return 2
-    else:
+
+    if n_jobs == 0:
+        raise ValueError("n_jobs == 0 in Parallel has no meaning")
+
+    if not n_jobs:
         if nx.config.backends.parallel.active:
             n_jobs = nx.config.backends.parallel.n_jobs
         else:
             from joblib.parallel import get_active_backend
 
             n_jobs = get_active_backend()[1]
-        n_cpus = os.cpu_count()
-        if n_jobs is None:
-            return 1
-        if n_jobs < 0:
-            return n_cpus + n_jobs + 1
-        if n_jobs == 0:
-            raise ValueError("n_jobs == 0 in Parallel has no meaning")
-        return int(n_jobs)
-
-
-def create_iterables(G, iterator, n_cores, list_of_iterator=None):
-    """Creates an iterable of function inputs for parallel computation
+
+    if not n_jobs:
+        return 1  # Default to 1 if no valid n_jobs is found or passed
+    if n_jobs < 0:
+        return os.cpu_count() + n_jobs + 1
+
+    return int(n_jobs)
+
+
+def create_iterables(
+    G,
+    iterator,
+    n_cores,
+    list_of_iterator=None,
+):
+    """
+    Create an iterable of function inputs for parallel computation
     based on the provided iterator type.
 
     Parameters
-    -----------
-    G : NetworkX graph
-    iterator : str
-        Type of iterator. Valid values are 'node', 'edge', 'isolate'
-
-    Returns:
-    --------
-    iterable : Iterable
+    ----------
+    G : networkx.Graph
+        The NetworkX graph.
```

Comment:
I would prefer not to put a specific class type here because it could be a DiGraph or MultiGraph, etc. How about `NetworkX graph`?

Reply:
Sounds good, changed